rotkehlchen/db/calendar.py

```diff
@@ -0,0 +1,219 @@
+    timestamp: Timestamp
+    description: str | None
+    counterparty: str | None
+    address: BlockchainAddress | None
+
+    def serialize(self) -> dict[str, Any]:
+        return {
+            'identifier': self.identifier,
+            'name': self.name,
+            'description': self.description,
+            'counterparty': self.counterparty,
+            'timestamp': self.timestamp,
+            'address': self.address,
+        }
+
+
+class CalendarFilterQuery(DBFilterQuery, FilterWithTimestamp):
+    """Filter used to query calendar events"""
+
+    @classmethod
+    def make(
+            cls: type['CalendarFilterQuery'],
+            and_op: bool = True,
+            order_by_rules: list[tuple[str, bool]] | None = None,
+            limit: int | None = None,
+            offset: int | None = None,
+            from_ts: Timestamp | None = None,
+            to_ts: Timestamp | None = None,
+            addresses: list[BlockchainAddress] | None = None,
+            name: str | None = None,
+            description: str | None = None,
+            counterparty: str | None = None,
+            identifiers: list[int] | None = None,
+    ) -> 'CalendarFilterQuery':
+        if order_by_rules is None:
+            order_by_rules = [('timestamp', True)]
+
+        filter_query = cls.create(
+            and_op=and_op,
+            limit=limit,
+            offset=offset,
+            order_by_rules=order_by_rules,
+        )
+        filter_query.timestamp_filter = DBTimestampFilter(
+            and_op=True,
+            from_ts=from_ts,
+            to_ts=to_ts,
+        )
+
+        filters: list['DBFilter'] = [filter_query.timestamp_filter]
+        if addresses is not None:
+            filters.append(DBMultiStringFilter(
+                and_op=True,
+                column='address',
+                values=addresses,
+            ))
+        if name is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='name',
+                search_string=name,
+            ))
+        if description is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='description',
+                search_string=description,
+            ))
+        if counterparty is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='counterparty',
+                search_string=counterparty,
+            ))
+        if identifiers is not None:
+            filters.append(DBMultiIntegerFilter(
+                and_op=True,
+                column='identifier',
+                values=identifiers,
+            ))
+
+        filter_query.filters = filters
+        return filter_query
+
+
+class DBCalendar:
+
+    def __init__(self, database: 'DBHandler') -> None:
+        self.db = database
+
+    def create_calendar_entry(
+            self,
+            name: str,
+            timestamp: int,
```

Comment:
again same concerns for type of timestamp and description as before. Entire file